src/Resources/contao/drivers/DC_Folder.php

```diff
@@ -178,6 +178,26 @@ public function __construct($strTable)
 			$this->arrValidFileTypes = \StringUtil::trimsplit(',', strtolower($GLOBALS['TL_DCA'][$this->strTable]['config']['validFileTypes']));
 		}
 
+		// Configure the picker
```

Comment:
Obviously I would not make it `abstract`. I'd just add the "normal" case to `DataContainer` and override the stuff that differs in the child tables :-)